### PR TITLE
Merge wingo-next

### PR DESCRIPTION
**Pull request by wingo**
This reconciles some of our upstream changes with those from master, in prep for a merge.  Note that this removes the ingress drop monitor on the NFV; perhaps it will come back in SnabbCo#1045.

### CODE REVIEW

File: src/doc/performance-tuning.md [hunk 1]
```diff
@@ -1,91 +1,189 @@
...
 
 ```
-GRUB_CMDLINE_LINUX_DEFAULT="... default_hugepagesz=1GB hugepagesz=1G hugepages=64 ..."
+boot.kernelParams = [ "default_hugepagesz=2048K" "hugepagesz=2048K"
+                      "hugepages=10000" "isolcpus=1-5,7-11" ];
 ```
+
+For Ubuntu, edit the Grub configuration as described above.
+
+To use one-gigabyte huge pages, replace `2048K` above with `1G1 and
```
Comment (dpino) [inline src/doc/performance-tuning.md:183]: typo s/1G1/1G/

File: src/doc/performance-tuning.md [hunk 2]
```diff
@@ -110,95 +209,189 @@ Node 1 HugePages_Total:    32
...
-Memory shared between Snabb (by means of huge page mapping) and the NIC must share the same node. This is achieved via 'numactl', once the correct node is identified for a given NIC port/PCI address. 
-To find the node for a given PCI address, use cpulistaffinity combined with numactl (TODO: is there a more direct way??):
+### Tweak ring buffer sizes
+
+The way that Snabb interfaces with a NIC is that it will configure the
+NIC to receive incoming packets into a /ring buffer/.  This ring buffer
+is allocated by Snabb (incidentally, to a huge page; see above) and will
+be filled by the NIC.  It has to be a power of 2 in size: so it can hold
+space for 64 packets, 128 packets, 256 packets, and so on.  The default
+size is 512 packets and the maximum is 65536.  The NIC will fill this
```
Comment (dpino) [inline src/doc/performance-tuning.md:291]: Default num_descriptors has been varying every now and then. Right now in master the default is 1024.

**Comment by dpino** [comment]
LGTM.

**Comment by wingo** [comment]
Thanks for review.  I'll fix the doc upstream.

**Pull request merged**